sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/ExpressionConverter.java

```diff
@@ -920,8 +946,23 @@ private RexNode convertResolvedFunctionCall(
       }
 
       for (ResolvedExpr expr : functionCall.getArgumentList()) {
-        operands.add(convertRexNodeFromResolvedExpr(expr, columnList, fieldList));
+        operands.add(
+            convertRexNodeFromResolvedExpr(expr, columnList, fieldList, outerFunctionArguments));
+      }
+    } else if (funGroup.equals(USER_DEFINED_FUNCTIONS)) {
```

Comment:
Do you think for Java UDF, will this code path help?

Reply:
I am pretty sure we will need it, unless we can somehow avoid ExpressionConverter for Java UDF?

Reply:
I see. For Java UDF (without a nested call), it might not go through the process to convert every of its arguments. But for nested call cases, especially with builtin functions, it could go through this process.

We can keep current implementation in this PR now.